### PR TITLE
fix(agent): Disable profile generator -> speed up agent setup

### PR DESCRIPTION
**Pull request by Pwuts**
## **User description**
- **Remove dead `AIProfile.api_budget` attribute**
- **Generalize example call in profile_generator.py**
- **Remove `agent.ai_profile` and `agent.directives` attributes**
- **Add default AutoGPT profile to ai_profile.py & disable profile generator**
- **Disable custom AI profile generation in agent_protocol_server.py**

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Refactored `create_agent` to handle optional AIProfile, using defaults when not provided.
- Generalized the example call in `profile_generator.py` for broader use.
- Streamlined agent initialization by removing specific AI profiles and directives.
- Disabled budget constraints in system prompts to avoid unnecessary restrictions.
- Optimized agent creation in the protocol server by replacing `generate_agent_for_task` with `create_agent`.
- Replaced task-specific AI profile generation with default profiles in the main application flow.
- Simplified the `SystemComponent` by removing unused directives related to AI profiles.
- Updated `AIProfile` to have default values and removed the `api_budget` attribute.
- Adjusted unit tests to align with the removal of `api_budget` from `AIProfile`.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>configurators.py</strong><dd><code>Refactor create_agent to Handle Optional AIProfile</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/agent_factory/configurators.py

<li>Made <code>ai_profile</code> parameter optional in <code>create_agent</code> function, <br>defaulting to a new <code>AIProfile</code> instance if not provided.<br> <li> Added handling for missing <code>ai_profile</code> and <code>directives</code> by creating <br>default instances within the function.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-989f831e9a5aa2e4e47738ecf38e238cb5f42842f5366df9fb480508cfc5d50f">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>profile_generator.py</strong><dd><code>Simplify and Generalize Example Call in Profile Generator</code></dd></summary>
<hr>
      
autogpts/autogpt/autogpt/agent_factory/profile_generator.py

<li>Simplified the structure of <code>_example_call</code> to a more generalized <br>format.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-6277b2e108abafccb61ea03582ba6596db71fc9272d4f573da9967e9ecd9cbfd">+40/-46</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>agent.py</strong><dd><code>Streamline Agent Initialization by Removing Specific Profiles</code></dd></summary>
<hr>
      
autogpts/autogpt/autogpt/agents/agent.py

<li>Removed <code>ai_profile</code> and <code>directives</code> from agent initialization, using <br>system-wide defaults instead.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-9224bfb0fac9d6e9ef7e4533bacf7dee4f89ba02339088c95f5e7cfc3455dd94">+1/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>agent_protocol_server.py</strong><dd><code>Optimize Agent Creation in Protocol Server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/agent_protocol_server.py

<li>Replaced <code>generate_agent_for_task</code> with <code>create_agent</code> to streamline agent <br>creation.<br> <li> Removed logging for agent creation to potentially improve performance.<br> <br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-8687cb793322c5a9217f5182d796e3befbcb6c37387f5614f719c514874d5a3b">+5/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Use Default AI Profiles in Main Application Flow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/app/main.py

<li>Replaced task-specific AI profile generation with a default profile <br>setup.<br> <li> Added comments to reconsider the performance benefits of task-oriented <br>profiles.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-3ca14b0308a80bf8ccf59bfce0ab9fd9ec4bc8223b14493a24f57b9150e704de">+17/-9</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>system.py</strong><dd><code>Simplify System Component by Removing Unused Directives</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/commands/system.py

<li>Removed <code>DirectiveProvider</code> implementation and related methods as <br><code>ai_profile</code> is no longer used here.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-a1c059daafd66c52fbaa44e371c84abf678cc10c044bb7fdd4308e72b1844fd1">+3/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ai_profile.py</strong><dd><code>Update AIProfile with Defaults and Remove Budget</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/config/ai_profile.py

<li>Set default values for <code>ai_name</code> and <code>ai_role</code> in <code>AIProfile</code>.<br> <li> Removed <code>api_budget</code> attribute from <code>AIProfile</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-dc7c306115138cd8498f36afaae3a24e9cfcd7bd9efb310d9456dcc52037d14e">+15/-9</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Bug_fix
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>one_shot.py</strong><dd><code>Disable Budget Constraint in System Prompt</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/autogpt/agents/prompt_strategies/one_shot.py

- Commented out the budget constraint addition to the system prompt.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-c8c3d9b541659d0906f0e03a0b6ed211f5aa3f243393556a9d711ec4a908d193">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test_ai_profile.py</strong><dd><code>Update AIProfile Unit Tests for Budget Removal</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpts/autogpt/tests/unit/test_ai_profile.py

<li>Updated tests to reflect the removal of <code>api_budget</code> from <code>AIProfile</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-b5d3129efa3c68fd8d2745c5cf9d6f251b7c46adcd47dbb4ceece9b02cff598e">+0/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Made `ai_profile` parameter optional in `create_agent` function and added default initialization.
- Generalized the example call in `AgentProfileGeneratorConfiguration` and simplified its structure.
- Removed `ai_profile` and `directives` attributes from `Agent` class and updated `SystemComponent` initialization.
- Commented out budget constraint generation in `build_system_prompt`.
- Replaced `generate_agent_for_task` with `create_agent` in `create_task` and disabled custom AI profile generation.
- Removed task-specific AI profile generation and used default `AIProfile` and `AIDirectives` in `run_auto_gpt`.
- Removed `AIProfile` and budget constraints from `SystemComponent`.
- Added default values for `ai_name` and `ai_role` in `AIProfile` and removed `api_budget` attribute.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>configurators.py</strong><dd><code>Make `ai_profile` parameter optional in `create_agent`</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpt/autogpt/agent_factory/configurators.py

<li>Made <code>ai_profile</code> parameter optional in <code>create_agent</code> function.<br> <li> Added default <code>AIProfile</code> initialization if not provided.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-d5694ae518ed98ecd6e39c1e58d996e52c371904f128c25e9723eb66e542356d">+2/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>profile_generator.py</strong><dd><code>Generalize example call in `AgentProfileGeneratorConfiguration`</code></dd></summary>
<hr>
      
autogpt/autogpt/agent_factory/profile_generator.py

<li>Generalized the example call in <code>AgentProfileGeneratorConfiguration</code>.<br> <li> Simplified the structure of <code>_example_call</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-90bd68b6e67084b6e8ea73bb3d2d4557717c00577c12d141d52d0d9d28e95023">+40/-46</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>agent.py</strong><dd><code>Remove unused `ai_profile` and `directives` attributes from `Agent`</code></dd></summary>
<hr>
      
autogpt/autogpt/agents/agent.py

<li>Removed <code>ai_profile</code> and <code>directives</code> attributes from <code>Agent</code> class.<br> <li> Updated <code>SystemComponent</code> initialization to remove unused directives.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-e1d4e4fb86c09b652ada361f79901d99c13104cea5226264a39516e7df041760">+1/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>one_shot.py</strong><dd><code>Disable budget constraint generation in system prompt</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
autogpt/autogpt/agents/prompt_strategies/one_shot.py

<li>Commented out budget constraint generation in <code>build_system_prompt</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-8c61126a8cf1e5e161f76dc915c1d16282a92304ac3a15b0ec5c836197bfae36">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>agent_protocol_server.py</strong><dd><code>Replace <code>generate_agent_for_task</code> with <code>create_agent</code> in <code>create_task</code></code></dd></summary>
<hr>
      
autogpt/autogpt/app/agent_protocol_server.py

<li>Replaced <code>generate_agent_for_task</code> with <code>create_agent</code> in <code>create_task</code>.<br> <li> Disabled custom AI profile generation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-2b44189224da3c14ee0517c56aa3cbd3c67fa2a3d04beca26246164231b28561">+5/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.py</strong><dd><code>Remove task-specific AI profile generation in main application flow</code></dd></summary>
<hr>
      
autogpt/autogpt/app/main.py

<li>Removed task-specific AI profile generation.<br> <li> Used default <code>AIProfile</code> and <code>AIDirectives</code> in <code>run_auto_gpt</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-b6333c164563761ad25a097cf17cc52715d010d89e346cad5166553aee1e72a5">+27/-12</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>system.py</strong><dd><code>Remove `AIProfile` and budget constraints from `SystemComponent`</code></dd></summary>
<hr>
      
forge/forge/components/system/system.py

- Removed `AIProfile` and budget constraints from `SystemComponent`.

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-d0162a09e490a9f5a3c7a43ca7d3ddd28624b62c0c1a9639ca74e8b585e3024d">+0/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ai_profile.py</strong><dd><code>Add default values and remove `api_budget` in `AIProfile`</code></dd></summary>
<hr>
      
forge/forge/config/ai_profile.py

<li>Added default values for <code>ai_name</code> and <code>ai_role</code> in <code>AIProfile</code>.<br> <li> Removed <code>api_budget</code> attribute from <code>AIProfile</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7121/files#diff-d38670f5ea75088318c4ce677e5b1212ba09bbe429d737f677c2d9d37b9e46a0">+12/-3</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: autogpt/autogpt/agents/prompt_strategies/one_shot.py [hunk 1]
```diff
@@ -162,7 +162,7 @@ def build_system_prompt(
                 self.config.body_template.format(
                     constraints=format_numbered_list(
                         ai_directives.constraints
-                        + self._generate_budget_constraint(ai_profile.api_budget)
+                        # + self._generate_budget_constraint(ai_profile.api_budget)
```
Comment (ntindle) [inline autogpt/autogpt/agents/prompt_strategies/one_shot.py]: nit: remove

**Review by ntindle — DISMISSED** [review]

File: autogpt/autogpt/agents/prompt_strategies/one_shot.py (hunk 1 above)
Comment (Pwuts) [reply autogpt/autogpt/agents/prompt_strategies/one_shot.py]: done

**Review by ntindle — APPROVED** [review]

**Pull request merged**